### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by jsancio**

### CODE REVIEW

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala [hunk 1]
```diff
@@ -124,42 +134,72 @@ private[cosmos] final class Cosmos(
...
+        :+: packageInstall.map(jsonToResponse)
+        :+: packageList.map(jsonToResponse)
+        :+: packageListVersions.map(jsonToResponse)
+        :+: packageRender.map(jsonToResponse)
+        :+: packageRepositoryAdd.map(jsonToResponse)
+        :+: packageRepositoryDelete.map(jsonToResponse)
+        :+: packageRepositoryList.map(jsonToResponse)
+        :+: packageSearch.map(jsonToResponse)
+        :+: packageUninstall.map(jsonToResponse)
+        :+: serviceStart.map(jsonToResponse)
```
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: We need to fix this. We need to either implement a finch encoder for circe's `Json` or turn all of the `Endpoint[Json]`s into `Endpoint[Response]`.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala [hunk 2]
```diff
@@ -172,40 +212,64 @@ private[cosmos] final class Cosmos(
...
-        .serve(iface, getService(s"srv/$name"))
+  private[this] def startServer(): Option[ListeningServer] = {
+    // TODO: Make default http port configurable
+    Some(new InetSocketAddress("0.0.0.0", 7070)).map { iface => // scalastyle:ignore magic.number
+      //.configured(Label(name)) TODO: Add this back when we find out what it is doing.
+      //.configured(Http.param.MaxRequestSize(config.maxRequestSize.megabytes)) TODO: Add this back
+
+      Http
+        .server
+        .serve(iface, startCosmos(NullStatsReceiver)) // TODO: User a better stats receiver
```
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: Clean this up.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala [hunk 3]
```diff
@@ -172,40 +212,64 @@ private[cosmos] final class Cosmos(
...
+
+  def main(): Unit = {
+    for (httpServer <- startServer) {
+      logger.info(s"HTTP server started on ${httpServer.boundAddress}")
+      closeOnExit(httpServer)
+      Await.result(httpServer)
+    }
 
-    startCosmos(statsReceiver.scope("cosmos"))
+    // TODO: Start HTTPS server
```
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: Make sure that we listen to HTTPS if required and listen to the admin port.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala [hunk 4]
```diff
@@ -172,40 +212,64 @@ private[cosmos] final class Cosmos(
...
+        val engineFactory = () =>
+          Ssl.server(certificatePath, keyPath, null, null, null) // scalastyle:ignore null
+
+        //.configured(Label(name)) TODO: Add this back when we find out what it is doing
+        //.configured(Http.param.MaxRequestSize(config.maxRequestSize.megabytes)) TODO: Add this back
+
+        Http
+          .server
+          .withTls(Netty3ListenerTLSConfig(engineFactory))
+          .serve(iface, startCosmos(NullStatsReceiver)) // TODO: User a better stats receiver
```
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: Enable this and clean it up.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala [hunk 5]
```diff
@@ -366,4 +430,18 @@ object Cosmos extends FinchServer {
...
+      l.setLevel(Level.OFF)
+    }
+    org.slf4j.LoggerFactory.getLogger("slf4j-logging").debug("Installing SLF4JLogging")
+    SLF4JBridgeHandler.install()
+  }
+
+  onExit {
+    org.slf4j.LoggerFactory.getLogger("slf4j-logging").debug("Uninstalling SLF4JLogging")
+    SLF4JBridgeHandler.uninstall()
+  }
```
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: Move this to a helper class.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/CosmosRepository.scala [hunk 6]
```diff
@@ -30,19 +29,31 @@ trait CosmosRepository extends PackageCollection {
...
 
-final class DefaultCosmosRepository(
-    override val repository: rpc.v1.model.PackageRepository,
-    universeClient: UniverseClient,
-    clock: Clock = new LowResClock(Amount.of(1L, Time.SECONDS))
+// TODO: Rename DefaultCosmosRepository to CosmosRepository
+private[repository] final class DefaultCosmosRepository (
+  override val repository: rpc.v1.model.PackageRepository,
+  universeClient: UniverseClient,
+  clock: Clock
```
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/CosmosRepository.scala]: Rename DefaultCosmosRepository to just CosmosRepository. I think we don't need this to be an interface.

File: cosmos-server/src/test/scala/com/mesosphere/cosmos/circe/EncodersDecodersSpec.scala [hunk 7]
```diff
@@ -109,34 +112,34 @@ class EncodersDecodersSpec extends FreeSpec {
 
   "Encoder for io.circe.Error" - {
     "DecodingFailure should specify path instead of cursor history" - {
-      "when value is valid type but not acceptable" in {
-        val Xor.Left(err: DecodingFailure) = loadAndDecode(
+      "when value is valid type but not acceptable" taggedAs Tag("TODO: file issue") ignore {
```
Comment (jsancio) [inline cosmos-server/src/test/scala/com/mesosphere/cosmos/circe/EncodersDecodersSpec.scala]: Create an issue for this and update the tag to point to it.

**Comment by cruhland** [comment]
@jsancio I just merged #556 if you'd like to pick up the changes. I'll hold off on reviewing this until you say it's ready.

**Comment by jsancio** [comment]
Yes. Give me a couple of days and I'll ping you when it is ready for review.

**Comment by jsancio** [comment]
@cruhland The integration tests should pass now. You can start reviewing this if you like. Ignore the `Cosmos.scala`, I am still working on that.

File: cosmos-finch/src/main/scala/com/mesosphere/cosmos/finch/RequestError.scala [hunk 8]
```diff
@@ -1,11 +1,35 @@
...
 import io.circe.JsonObject
+import io.finch.Error
+import io.finch.items.HeaderItem
 
-abstract class RequestError(causedBy: Option[Throwable] = None) extends RuntimeException(causedBy.orNull) {
-  def status: Status
-  def errType: String
-  def getData: Option[JsonObject]
-  def getHeaders: Map[String, String]
+object RequestError {
```
Comment (jsancio) [inline cosmos-finch/src/main/scala/com/mesosphere/cosmos/finch/RequestError.scala]: Make this a `package object finch`.

File: cosmos-finch/src/main/scala/com/mesosphere/cosmos/finch/EndpointHandler.scala [hunk 9]
```diff
@@ -14,7 +14,7 @@ abstract class EndpointHandler[Request, Response](successStatus: Status = Status
       val encodedResponse = response.asJson(context.responseEncoder.encoder)
 
       Output.payload(encodedResponse, successStatus)
-        .withContentType(Some(context.responseEncoder.mediaType.show))
+        .withHeader("Content-Type" -> context.responseEncoder.mediaType.show)
```
Comment (cruhland) [inline cosmos-finch/src/main/scala/com/mesosphere/cosmos/finch/EndpointHandler.scala]: I've been importing `com.twitter.finagle.http.Fields` and using `Fields.ContentType` when I need the string header name.

File: cosmos-finch/src/test/scala/com/mesosphere/cosmos/finch/EndpointHandlerSpec.scala [hunk 10]
```diff
@@ -165,14 +170,13 @@ object EndpointHandlerSpec {
...
-    val Xor.Right(body) = output.value.as[A]
+    val Right(body) = output.value.as[A]
     body
   }
 
   def extractContentType(result: Future[Output[Json]]): String = {
     val output = Await.result(result)
-    val Some(contentType) = output.contentType
-    contentType
+    output.headers("Content-Type")
```
Comment (cruhland) [inline cosmos-finch/src/test/scala/com/mesosphere/cosmos/finch/EndpointHandlerSpec.scala]: `Fields.ContentType`?

File: cosmos-finch/src/test/scala/com/mesosphere/cosmos/finch/RequestValidatorsSpec.scala [hunk 11]
```diff
@@ -332,12 +333,13 @@ object RequestValidatorsSpec {
     Gen.listOf(Gen.frequency((10, Gen.alphaNumChar), Gen.freqTuple((1, '/')))).map(_.mkString)
   }
 
-  def validate[Req, Res](
+  def validateOuput[Req, Res](
```
Comment (cruhland) [inline cosmos-finch/src/test/scala/com/mesosphere/cosmos/finch/RequestValidatorsSpec.scala]: Typo, should be `validateOutput`.

File: cosmos-finch/src/test/scala/com/mesosphere/cosmos/finch/RequestValidatorsSpec.scala [hunk 12]
```diff
@@ -70,8 +70,9 @@ final class RequestValidatorsSpec extends FreeSpec with Matchers with PropertyCh
       request: HttpRequest,
       validator: Endpoint[EndpointContext[Req, Res]]
     ): EndpointContext[Req, Res] = {
-      val Some((_, eval)) = validator(Input(HttpRequest.toFinagle(request)))
-      Await.result(eval.value).value
+      val finagleRequest = HttpRequest.toFinagle(request)
+      val Some((_, eval)) = validator(Input(finagleRequest, finagleRequest.path.split("/")))
+      Await.result(eval.run).value
```
Comment (cruhland) [inline cosmos-finch/src/test/scala/com/mesosphere/cosmos/finch/RequestValidatorsSpec.scala]: Delegate to `validateOutput`?

```scala
validateOutput(validator, request).get().value
```

File: cosmos-finch/src/test/scala/com/mesosphere/cosmos/finch/VersionedResponsesSpec.scala [hunk 13]
```diff
@@ -70,12 +77,12 @@ object VersionedResponsesSpec {
       .setHeader("Content-Type", applicationJson.show)
       .buildPost(Buf.Utf8(body))
 
-    Input(request)
+    Input(request, request.path.split("/"))
```
Comment (cruhland) [inline cosmos-finch/src/test/scala/com/mesosphere/cosmos/finch/VersionedResponsesSpec.scala]: I've seen this `request.path.split("/")` pattern several times...would it be worth it to create a utility method `http.Request => Input`?

File: cosmos-json/src/main/scala/com/mesosphere/cosmos/storage/v1/circe/Decoders.scala [hunk 14]
```diff
@@ -1,21 +1,23 @@
...
+import com.mesosphere.cosmos.storage.v1.model.PendingStatus
 import com.mesosphere.cosmos.storage.v1.model.Uninstall
 import com.mesosphere.cosmos.storage.v1.model.UniverseInstall
 import com.mesosphere.universe.v3.circe.Decoders._
 import io.circe.Decoder
 import io.circe.DecodingFailure
 import io.circe.HCursor
 import io.circe.generic.semiauto._
+import scala.util.Left
+import scala.util.Right
```
Comment (cruhland) [inline cosmos-json/src/main/scala/com/mesosphere/cosmos/storage/v1/circe/Decoders.scala]: The `Right` doesn't seem to be used?

File: cosmos-json/src/main/scala/com/mesosphere/universe/v3/circe/Decoders.scala [hunk 15]
```diff
@@ -80,8 +86,9 @@ object Decoders {
 
   implicit val decodeV3PackagingVersion: Decoder[PackagingVersion] = Decoder.instance[PackagingVersion] { (c: HCursor) =>
     c.as[String].map(PackagingVersion(_)).flatMap {
-      case Return(v) => Xor.Right(v)
-      case Throw(e) => Xor.Left(DecodingFailure(e.getMessage, c.history))
+      case Return(v) => Right(v)
+      case Throw(e) =>
+        Left(DecodingFailure(e.getMessage, c.history))
```
Comment (cruhland) [inline cosmos-json/src/main/scala/com/mesosphere/universe/v3/circe/Decoders.scala]: Nitpick: make this `case` all on one line, for consistency

Comment (cruhland) [inline cosmos-json/src/main/scala/com/mesosphere/universe/v3/circe/Decoders.scala]: Alternatively, wrap both lines

File: cosmos-render/src/main/scala/com/mesosphere/cosmos/render/PackageDefinitionRenderer.scala [hunk 16]
```diff
@@ -51,9 +59,15 @@ object PackageDefinitionRenderer {
...
+                    obj => JsonObject.singleton("labels", Json.fromJsonObject(obj))
+                  ),
+                  Some(existingLabels).map(
+                    obj => JsonObject.singleton("labels", Json.fromJsonObject(obj))
+                  ),
+                  Some(labels.nonOverridableLabelsJson).map(
+                    obj => JsonObject.singleton("labels", Json.fromJsonObject(obj))
+                  ),
                   marathonAppId.map(appIdDoc)
                 ).flatten.foldLeft(JsonObject.empty)(merge)
```
Comment (cruhland) [inline cosmos-render/src/main/scala/com/mesosphere/cosmos/render/PackageDefinitionRenderer.scala]: This might be even cleaner:

```scala
val labels = MarathonLabels(pkgDef, sourceUri)
val newLabels: Seq[Option[JsonObject]] =
  Seq(labels.requiredLabelsJson, existingLabels, labels.nonOverridableLabelsJson)
    .map(obj => Some(JsonObject.singleton("labels", Json.fromJsonObject(obj))))

val newLabelsAndAppId = (newLabels :+ marathonAppId.map(appIdDoc))
  .flatten
  .foldLeft(JsonObject.empty)(merge)
```

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala [hunk 17]
```diff
@@ -140,24 +161,44 @@ private[cosmos] final class Cosmos(
...
     val api = endpoints.handle {
-      case re: RequestError =>
+      case re: CosmosError =>
         stats.counter(s"definedError/${sanitiseClassName(re.getClass)}").incr()
-        val output = Output.failure(re, re.status).withContentType(Some(MediaTypes.ErrorResponse.show))
+        val output = Output.failure(
+          re,
+          re.status
+        ).withHeader(
+          "Content-Type" -> MediaTypes.ErrorResponse.show
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: `Fields.ContentType`?

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala [hunk 18]
```diff
@@ -140,24 +161,44 @@ private[cosmos] final class Cosmos(
...
+        )
         re.getHeaders.foldLeft(output) { case (out, kv) => out.withHeader(kv) }
-      case fe: _root_.io.finch.Error =>
+      case fe @ (_: _root_.io.finch.Error | _: _root_.io.finch.Errors) =>
         stats.counter(s"finchError/${sanitiseClassName(fe.getClass)}").incr()
-        Output.failure(fe, Status.BadRequest).withContentType(Some(MediaTypes.ErrorResponse.show))
+        Output.failure(
+          fe.asInstanceOf[Exception], // Must be an Exception based on the types
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: Why doesn't Scala just infer the type?

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala [hunk 19]
```diff
@@ -140,24 +161,44 @@ private[cosmos] final class Cosmos(
...
+          Status.BadRequest
+        ).withHeader(
+          "Content-Type" -> MediaTypes.ErrorResponse.show
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: `Fields.ContentType`?

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala [hunk 20]
```diff
@@ -140,24 +161,44 @@ private[cosmos] final class Cosmos(
...
+        )
       case e: Exception if !e.isInstanceOf[_root_.io.finch.Error] =>
         stats.counter(s"unhandledException/${sanitiseClassName(e.getClass)}").incr()
         logger.warn("Unhandled exception: ", e)
-        Output.failure(e, Status.InternalServerError).withContentType(Some(MediaTypes.ErrorResponse.show))
+        Output.failure(
+          e,
+          Status.InternalServerError
+        ).withHeader(
+          "Content-Type" -> MediaTypes.ErrorResponse.show
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: `Fields.ContentType`?

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala [hunk 21]
```diff
@@ -140,24 +161,44 @@ private[cosmos] final class Cosmos(
...
+        )
       case t: Throwable if !t.isInstanceOf[_root_.io.finch.Error] =>
         stats.counter(s"unhandledThrowable/${sanitiseClassName(t.getClass)}").incr()
         logger.warn("Unhandled throwable: ", t)
-        Output.failure(new Exception(t), Status.InternalServerError).withContentType(Some(MediaTypes.ErrorResponse.show))
+        Output.failure(
+          new Exception(t),
+          Status.InternalServerError
+        ).withHeader(
+          "Content-Type" -> MediaTypes.ErrorResponse.show
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: `Fields.ContentType`?

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala [hunk 22]
```diff
@@ -140,24 +161,44 @@ private[cosmos] final class Cosmos(
...
+      case fe @ (_: _root_.io.finch.Error | _: _root_.io.finch.Errors) =>
         stats.counter(s"finchError/${sanitiseClassName(fe.getClass)}").incr()
-        Output.failure(fe, Status.BadRequest).withContentType(Some(MediaTypes.ErrorResponse.show))
+        Output.failure(
+          fe.asInstanceOf[Exception], // Must be an Exception based on the types
+          Status.BadRequest
+        ).withHeader(
+          "Content-Type" -> MediaTypes.ErrorResponse.show
+        )
       case e: Exception if !e.isInstanceOf[_root_.io.finch.Error] =>
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: The guard clause (`if !e.isInstanceOf[_root_.io.finch.Error]`) shouldn't be needed, since that case is handled above

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala [hunk 23]
```diff
@@ -140,24 +161,44 @@ private[cosmos] final class Cosmos(
...
         stats.counter(s"unhandledException/${sanitiseClassName(e.getClass)}").incr()
         logger.warn("Unhandled exception: ", e)
-        Output.failure(e, Status.InternalServerError).withContentType(Some(MediaTypes.ErrorResponse.show))
+        Output.failure(
+          e,
+          Status.InternalServerError
+        ).withHeader(
+          "Content-Type" -> MediaTypes.ErrorResponse.show
+        )
       case t: Throwable if !t.isInstanceOf[_root_.io.finch.Error] =>
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: The guard clause (`if !t.isInstanceOf[_root_.io.finch.Error]`) shouldn't be needed, since that case is handled above

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/circe/Encoders.scala [hunk 24]
```diff
@@ -157,85 +127,26 @@ object Encoders extends LowPriorityImplicits {
...
-    new DerivedObjectEncoder[FieldType[K, H] :: T] {
-      override def encodeObject(a: FieldType[K, H] :: T): JsonObject = {
-        encodeTail.value.encodeObject(a.tail)
-      }
-    }
-  }
-
-  implicit val encodeCosmosError: Encoder[CosmosError] = deriveEncoder[CosmosError]
-
-
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/circe/Encoders.scala]: :100:

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/CosmosRepository.scala (hunk 6 above)
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/CosmosRepository.scala]: Agree, but this seems unrelated to the goal of the PR. Should we file an issue and defer this to another PR?

File: cosmos-server/src/test/scala/com/mesosphere/cosmos/repository/CosmosRepositorySpec.scala [hunk 25]
```diff
@@ -122,14 +122,11 @@ final class CosmosRepositorySpec extends FreeSpec {
...
-      assertResult("^\\Qminimal\\E$")(c.createRegex("minimal").toString)
-      assertResult("^\\Qmin\\E.*\\Qmal\\E$")(c.createRegex("min*mal").toString)
-      assertResult("^\\Qmini\\E.*\\Q.+\\E$")(c.createRegex("mini*.+").toString)
-      assertResult("^\\Qminimal\\E.*$")(c.createRegex("minimal*").toString)
-      assertResult("^\\Qminimal\\E.*.*$")(c.createRegex("minimal**").toString)
+      assertResult("^\\Qminimal\\E$")(CosmosRepository.createRegex("minimal").toString)
+      assertResult("^\\Qmin\\E.*\\Qmal\\E$")(CosmosRepository.createRegex("min*mal").toString)
+      assertResult("^\\Qmini\\E.*\\Q.+\\E$")(CosmosRepository.createRegex("mini*.+").toString)
+      assertResult("^\\Qminimal\\E.*$")(CosmosRepository.createRegex("minimal*").toString)
+      assertResult("^\\Qminimal\\E.*.*$")(CosmosRepository.createRegex("minimal**").toString)
```
Comment (cruhland) [inline cosmos-server/src/test/scala/com/mesosphere/cosmos/repository/CosmosRepositorySpec.scala]: These look related to the `CosmosRepository` changes, do we need them?

File: project/build.scala [hunk 26]
```diff
@@ -26,9 +26,10 @@ object CosmosBuild extends Build {
     val scalaUri = "0.4.11"
     val scalaTest = "2.2.4"
     val scalaCheck = "1.12.6"
-    val twitterUtilCore = "6.30.0"
+    val twitterUtilCore = "6.39.0"
     val zookeeper = "3.4.6"
     val aws = "1.11.63"
+    val slf4j = "1.7.10"
```
Comment (cruhland) [inline project/build.scala]: What do you think about alphabetizing the list?

Comment (cruhland) [inline project/build.scala]: Also, why did `slf4j` get added?

**Comment by cruhland** [comment]
@jsancio added some comments, waiting for further changes. No big concerns overall.

File: cosmos-finch/src/main/scala/com/mesosphere/cosmos/finch/RequestError.scala (hunk 8 above)
Comment (jsancio) [inline cosmos-finch/src/main/scala/com/mesosphere/cosmos/finch/RequestError.scala]: Done.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/CosmosRepository.scala (hunk 6 above)
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/CosmosRepository.scala]: Agree. I won't do it in this PR.

File: cosmos-finch/src/main/scala/com/mesosphere/cosmos/finch/EndpointHandler.scala (hunk 9 above)
Comment (jsancio) [inline cosmos-finch/src/main/scala/com/mesosphere/cosmos/finch/EndpointHandler.scala]: Done.

File: cosmos-finch/src/test/scala/com/mesosphere/cosmos/finch/EndpointHandlerSpec.scala (hunk 10 above)
Comment (jsancio) [inline cosmos-finch/src/test/scala/com/mesosphere/cosmos/finch/EndpointHandlerSpec.scala]: Done.

File: cosmos-finch/src/test/scala/com/mesosphere/cosmos/finch/RequestValidatorsSpec.scala (hunk 12 above)
Comment (jsancio) [inline cosmos-finch/src/test/scala/com/mesosphere/cosmos/finch/RequestValidatorsSpec.scala]: Maybe. Outside the scope of this PR.

File: cosmos-finch/src/test/scala/com/mesosphere/cosmos/finch/VersionedResponsesSpec.scala (hunk 13 above)
Comment (jsancio) [inline cosmos-finch/src/test/scala/com/mesosphere/cosmos/finch/VersionedResponsesSpec.scala]: Done.

File: cosmos-render/src/main/scala/com/mesosphere/cosmos/render/PackageDefinitionRenderer.scala (hunk 16 above)
Comment (jsancio) [inline cosmos-render/src/main/scala/com/mesosphere/cosmos/render/PackageDefinitionRenderer.scala]: I prefer to do this in a different PR. I need to investigate if we have a test for this code. I was just trying to change the formatting to be less than 100 characters.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala (hunk 18 above)
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: It infers `Throwable`. Not sure why.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala (hunk 22 above)
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: Done.

File: cosmos-json/src/main/scala/com/mesosphere/universe/v3/circe/Decoders.scala (hunk 15 above)
Comment (jsancio) [inline cosmos-json/src/main/scala/com/mesosphere/universe/v3/circe/Decoders.scala]: Done.

File: cosmos-server/src/test/scala/com/mesosphere/cosmos/repository/CosmosRepositorySpec.scala (hunk 25 above)
Comment (jsancio) [inline cosmos-server/src/test/scala/com/mesosphere/cosmos/repository/CosmosRepositorySpec.scala]: I did this because we were using a mocked clock to instantiate DefaultCosmosRepository just so we can create regex. Twitter common dropped support for mocked clocks so I removed the need to have to create a Cosmos repo just so we can create a regex.

File: project/build.scala (hunk 26 above)
Comment (jsancio) [inline project/build.scala]: We were getting his from finchserver. We need to include this ourselves now.

Comment (jsancio) [inline project/build.scala]: I'll alphabetize the list.

File: cosmos-render/src/main/scala/com/mesosphere/cosmos/render/PackageDefinitionRenderer.scala (hunk 16 above)
Comment (cruhland) [inline cosmos-render/src/main/scala/com/mesosphere/cosmos/render/PackageDefinitionRenderer.scala]: Oh ok, I was wondering why this changed. Never mind then.

**Comment by jsancio** [comment]
@cruhland Wohoo! This PR is ready for your review. Thanks!

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/package.scala [hunk 27]
```diff
@@ -1,8 +1,95 @@
...
+
+  def getKeyPath: Option[Path] = {
+    keyPath.getWithDefault.orElse(
+      _root_.io.github.benwhitehead.finch.keyPath.getWithDefault
+    )
+  }
+
+}
+
+package cosmos {
```
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/package.scala]: @cruhland I had to do this because `GlobalFlag`s are magical and only seem to work if namespaced with traditional Java packages and not with Scala's `package object`.

File: cosmos-server/src/main/resources/logback.xml [hunk 28]
```diff
@@ -34,10 +28,6 @@
     <appender-ref ref="NOEXCEPTION"/>
   </logger>
 
-  <logger name="access-log" level="trace" additivity="false">
-    <appender-ref ref="access-log"/>
-  </logger>
-
```
Comment (cruhland) [inline cosmos-server/src/main/resources/logback.xml]: Why were these lines removed?

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/package.scala (hunk 27 above)
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/package.scala]: Could you add your explanation as a comment?

Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/package.scala]: Also, I noticed that the `package object` worked with the flags in `io.github.benwhitehead.finch`, any idea why?

**Comment by cruhland** [comment]
@jsancio LGTM, I had a few questions though.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/package.scala (hunk 27 above)
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/package.scala]: Done. Added a comment.

Good catch. `io.github.benwhitehead.finch.*` wasn't working. I fixed it. I will test this with DC/OS before merging.

File: cosmos-server/src/main/resources/logback.xml (hunk 28 above)
Comment (jsancio) [inline cosmos-server/src/main/resources/logback.xml]: We don't need that logger and appender. That was for the logger coming from the finch server. Now we are using twitter's logger and don't need this additional configuration.

**Comment by jsancio** [comment]
@cruhland I had to make one small change to get it to work in `dcos-enterprise`. Check it out
